internals/cli/cli.go

```diff
@@ -191,61 +182,55 @@ func Parser(cli *client.Client) *flags.Parser {
 	}
 
 	// Add all commands
-	for _, c := range commands {
-		obj := c.Builder()
-		if x, ok := obj.(clientSetter); ok {
-			x.setClient(cli)
-		}
-		if x, ok := obj.(parserSetter); ok {
-			x.setParser(parser)
-		}
+	for _, cmdInfo := range commands {
+		obj := cmdInfo.New(&CmdOptions{Client: c, Parser: parser})
 
 		var target *flags.Command
-		if c.Debug {
+		if cmdInfo.Debug {
 			target = debugCmd
 		} else {
 			target = parser.Command
 		}
-		cmd, err := target.AddCommand(c.Name, c.Summary, strings.TrimSpace(c.Description), obj)
+		cmd, err := target.AddCommand(cmdInfo.Name, cmdInfo.Summary, strings.TrimSpace(cmdInfo.Description), obj)
```

Comment:
Same: why would we change short variable names unnecessarily and reuse them for different purposes inside a patch that is already a massive reorganization?  That creates deltas like the above, which are unnecessary noise in an otherwise good PR.